Tangem/Common/UI/Buttons/MainButton.swift

```diff
@@ -9,26 +9,40 @@
 import SwiftUI
 
 struct MainButton: View {
-    private let text: String
+    private let title: Title
     private let icon: Icon?
     private let style: Style
+    private let isLoading: Bool
     private let isDisabled: Bool
     private let action: () -> Void
 
     init(
-        text: String,
+        title: Title,
         icon: Icon? = nil,
         style: Style = .primary,
+        isLoading: Bool = false,
         isDisabled: Bool = false,
         action: @escaping (() -> Void)
     ) {
-        self.text = text
+        self.title = title
         self.icon = icon
         self.style = style
+        self.isLoading = isLoading
         self.isDisabled = isDisabled
         self.action = action
     }
 
+    init(settings: Settings) {
+        self.init(
+            title: settings.title,
+            icon: settings.icon,
+            style: settings.style,
+            isLoading: settings.isLoading,
+            isDisabled: settings.isDisabled,
+            action: settings.action
+        )
+    }
+
```

Comment:
может лучше сделать `title: LocalizedStringKey` и добавить инициализатор со `String`, в котором будем оборачивать строку в локализованную? Правда не знаю изменится ли скорость работы из-за того что это ключ локализации. Думал ещё сделать с дженериками, но чет кода ещё больше и не очень удобно юзать. Так на коленке быстро не получилось хорошего решения. Просто не особо нравится енум с string, key кейсами
```suggestion
init(
        title: String,
        icon: Icon? = nil,
        style: Style = .primary,
        isLoading: Bool = false,
        isDisabled: Bool = false,
        action: @escaping (() -> Void)
    ) {
        self.title = LocalizedStringKey(title)
        self.icon = icon
        self.style = style
        self.isLoading = isLoading
        self.isDisabled = isDisabled
        self.action = action
    }
```

Reply:
У LocalizedStringKey есть конструктор, который принимает не ключ, а просто строку текста. Можно через это делать.

Reply:
мы обсуждали этот вариант, но в таком случае обычная строка превращается в ключ локализации и тогда в некоторых случаях будет двойная работа. Например, мы с параметрами создаем строку, прогоняем её через локализацию `String(format:,CVarArg...)` передаем её в кнопку в виде строки и снова превращаем в ключ локализации, который ещё раз будет искаться.
Ты не знаешь на сколько затратно искать ключи локализации?

Reply:
Не знаю, но не думаю что это очень затратная работа. Там под капотом наверняка словарь.

И не думаю что там будет что-то искаться. Это разные конструкторы. Один создает текст из ключа, другой из захардкоженной строки.

https://developer.apple.com/documentation/swiftui/localizedstringkey/init(_:)
https://developer.apple.com/documentation/swiftui/localizedstringkey/init(stringliteral:)

Reply:
На всякий уточню — мы бы передавали в кнопку не строку String и потом ее бы конвертировали в LocalizedStringKey, а сразу бы уже передавали последний.

Reply:
можно правда сделать через `LocalizedStringKey/init(stringliteral:)`
я все грежу что мы сделаем генерацию, и перейдем на наш сгенерированный enum

Reply:
Было бы удобнее, думаю. После синка строк можно будет заняться.